manager/integration/tests/test_engine_upgrade.py

```diff
@@ -747,61 +749,261 @@ def test_engine_live_upgrade_with_intensive_data_writing(client, core_api, volum
+    wait_for_volume_healthy(client, volume1.name)
+    check_volume_data(volume1, data)
+
+    volume1 = client.by_id_volume(volume1.name)
+    volume2 = client.by_id_volume(volume2.name)
+    assert volume1.engineImage == default_img.image
+    assert volume2.engineImage == default_img.image
+    check_replica_engine(volume1, default_img.image)
+
+
+def test_auto_upgrade_engine_to_default_version_dr_volume(client, set_random_backupstore): # NOQA
+    """
+    Steps:
+
+    Preparation:
+    1. set up a backup store
+    2. Deploy a compatible new engine image
+
+    Test auto upgrade engine to default version in DR volume:
+    1. Create a backup for vol-1. Create a DR volume from the backup
+    2. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 3
+    3. Try to upgrade the DR volume engine's image to the new engine image
+    4. Verify that the Longhorn API returns error. Upgrade fails.
+    5. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 0
+    6. Try to upgrade the DR volume engine's image to the new engine image
+    7. Wait until the upgrade are completed (volumes' engine image changed,
        replicas' mode change to RW, reference count of the new engine image
        changed, engine and replicas' engine image changed)
-    7. Wait for the DR volume to finish restoring
-    8. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 3
-    9. In a 2-min retry loop, verify that Longhorn doesn't automatically
+    8. Wait for the DR volume to finish restoring
+    9. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 3
+    10. In a 2-min retry loop, verify that Longhorn doesn't automatically
        upgrade engine image for DR volume.
+    """
+    # Precondition
+    _, default_img_name, engine_upgrade_image, compatible_img, _ = \
+        prepare_auto_upgrade_engine_to_default_version(client)
+
+    volume1 = client.create_volume(name="vol-1", size=str(1 * Gi),
+                                   numberOfReplicas=REPLICA_COUNT)
+    volume1 = common.wait_for_volume_detached(client, volume1.name)
+    volume1.attach(hostId=get_self_host_id())
+    volume1 = wait_for_volume_healthy(client, volume1.name)
+
+    write_volume_random_data(volume1)
+
+    _, b, _, _ = create_backup(client, volume1.name)
+
+    common.cleanup_all_volumes(client)
+
+    # Test auto upgrade engine to default version in DR volume
+    dr_volume_name = "dr-expand-" + volume1.name
+    dr_volume = client.create_volume(name=dr_volume_name, size=SIZE,
+                                     numberOfReplicas=3, fromBackup=b.url,
+                                     frontend="", standby=True)
+
+    wait_for_backup_restore_completed(client, dr_volume_name, b.name)
+
+    dr_volume = common.wait_for_volume_healthy_no_frontend(client,
+                                                           dr_volume_name)
+    update_setting(client,
+                   "concurrent-automatic-engine-upgrade-per-node-limit",
```

Comment:
nit: suggest putting this in the [common.py](https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/common.py#L146-L180).

Reply:
Hi @c3y1huang , please to updated code, thank you